Sources/StreamChat/Controllers/ChannelListController/ChannelListController_Tests.swift

```diff
@@ -255,42 +255,28 @@ class ChannelListController_Tests: XCTestCase {
         AssertAsync.willBeEqual(controller.channels.map(\.cid), [cid])
     }
     
-    func test_newChannel_callsListHook() throws {
-        // Create wrapper test delegate
-        class TestLinkDelegate: ChatChannelListControllerDelegate {
-            let shouldListNewChannel: (ChatChannel) -> Bool
-            let shouldListUpdatedChannel: (ChatChannel) -> Bool
-            init(
-                shouldListNewChannel: @escaping (ChatChannel) -> Bool,
-                shouldListUpdatedChannel: @escaping (ChatChannel) -> Bool
-            ) {
-                self.shouldListNewChannel = shouldListNewChannel
-                self.shouldListUpdatedChannel = shouldListUpdatedChannel
-            }
-            
-            func controller(_ controller: ChatChannelListController, shouldAddNewChannelToList channel: ChatChannel) -> Bool {
-                shouldListNewChannel(channel)
-            }
-            
-            func controller(_ controller: ChatChannelListController, shouldListUpdatedChannel channel: ChatChannel) -> Bool {
-                shouldListUpdatedChannel(channel)
-            }
-        }
-        
-        // Simulate `synchronize` call
-        controller.synchronize()
+    func test_newChannel_callsListHook_whenSynchronized() throws {
```

Comment:
We should make sure this hook is not called before `sync` is called

Reply:
There's a `test_unlinkedChannels_doNotTriggerHooks_whenNotSynchronized`. 

And we can handle updates of linked channel even when a query is not yet synced. It is covered by `test_linkedChannels_doesTriggerUnlinkHook_whenNotSynchronized`